include/CL/opencl.hpp

```diff
@@ -10253,6 +10342,530 @@ namespace compatibility {
+
+    static CommandBufferKhr getDefault(cl_int * err = NULL)
+    {
+        std::call_once(default_initialized_, makeDefault);
+        detail::errHandler(default_error_, __CREATE_COMMAND_BUFFER_KHR_ERR);
+        if (err != NULL) {
+            *err = default_error_;
+        }
+        return default_;
+    }
+
+    static CommandBufferKhr setDefault(const CommandBufferKhr &default_cmd_buffer_khr)
+    {
+        std::call_once(default_initialized_, makeDefaultProvided, std::cref(default_cmd_buffer_khr));
+        detail::errHandler(default_error_);
+        return default_;
+    }
+
+    //! \brief Default constructor - initializes to NULL.
+    CommandBufferKhr() : detail::Wrapper<cl_type>() { }
+
+    explicit CommandBufferKhr(const vector<CommandQueue> &queues,
+        cl_command_buffer_properties_khr properties = 0,
+        cl_int* errcode_ret = NULL)
+    {
+        cl_int error = CL_SUCCESS;
+        cl_command_buffer_properties_khr command_buffer_properties[] = {
+            CL_COMMAND_BUFFER_FLAGS_KHR, properties, 0
+        };
+
+        /* initialization of addresses to extension functions */
+        error = initExtensions(queues[0].getInfo<CL_QUEUE_DEVICE>());
+
+        if (error == CL_SUCCESS)
+        {
+            object_ = pfn_clCreateCommandBufferKHR(queues.size(),
+                &queues[0](),
+                command_buffer_properties,
+                &error);
+        }
+
+        detail::errHandler(error, __CREATE_COMMAND_BUFFER_KHR_ERR);
+        if (errcode_ret != NULL) {
+            *errcode_ret = error;
+        }
+    }
+
+    explicit CommandBufferKhr(const cl_command_buffer_khr& commandBufferKhr, bool retainObject = false) :
+        detail::Wrapper<cl_type>(commandBufferKhr, retainObject) { }
+
+    CommandBufferKhr& operator = (const cl_command_buffer_khr& rhs)
+    {
+        detail::Wrapper<cl_type>::operator=(rhs);
+        return *this;
+    }
+
+    /*! \brief Copy constructor to forward copy to the superclass correctly.
+     * Required for MSVC.
+     */
+    CommandBufferKhr(const CommandBufferKhr& commandBufferKhr) : detail::Wrapper<cl_type>(commandBufferKhr) {}
+
+    /*! \brief Copy assignment to forward copy to the superclass correctly.
+     * Required for MSVC.
+     */
+    CommandBufferKhr& operator = (const CommandBufferKhr &commandBufferKhr)
+    {
+        detail::Wrapper<cl_type>::operator=(commandBufferKhr);
+        return *this;
+    }
+
+    /*! \brief Move constructor to forward move to the superclass correctly.
+     * Required for MSVC.
+     */
+    CommandBufferKhr(CommandBufferKhr&& commandBufferKhr) CL_HPP_NOEXCEPT_ : detail::Wrapper<cl_type>(std::move(commandBufferKhr)) {}
+
+    /*! \brief Move assignment to forward move to the superclass correctly.
+     * Required for MSVC.
+     */
+    CommandBufferKhr& operator = (CommandBufferKhr &&commandBufferKhr)
+    {
+        detail::Wrapper<cl_type>::operator=(std::move(commandBufferKhr));
+        return *this;
```

Comment:
Not clear why all these are necessary and the one synthesized by the compiler are not good.